django/middleware/csrf.py

```diff
@@ -436,7 +444,12 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
         if "HTTP_ORIGIN" in request.META:
             if not self._origin_verified(request):
```

Comment:
Could we restructure the code to avoid calling `_host_origin` twice (within `_origin_verified()` and then on line 451). So here could look like:

```suggestion
            host_origin = self._host_origin(request)
            if not self._origin_verified(request, host_origin):
```

And the error message could use `host_origin or "<disallowed host>"`.

Reply:
You still need to change line 451 to:

```
host_origin or "<disallowed host>",
```

Reply:
With @sarahboyce 's agreement that it should be a separate error code, it'll be fixed up a different way, but i have WIP code that does as you suggest. Working on it now.